test/controllers/comments_controller_test.rb

```diff
@@ -1,8 +1,42 @@
-require "test_helper"
+require 'test_helper'
 
 class CommentsControllerTest < ActionDispatch::IntegrationTest
-  test "should get create" do
-    get comments_create_url
-    assert_response :success
+  #setup user for comments
+  def setup
+    @user = users(:michael)
+    @comment = comments(:one)
+  end
+  # user can post a comment when logged in
+  test 'user can post a comment when logged in' do
+    get user_session_path
+    assert_equal 200, status
+    post user_session_path, params: { user: {
+                                              email: @user.email,
+                                              password: 'password'
+                                            }
+                                    }
+    follow_redirect!
+    assert_equal 200, status
+    get posts_path
+    assert_difference 'Comment.count', 1 do
+      post comments_path, params: { comment: { body: 'Lorem ipsum',
+                                               post_id: @comment.post.id,
+                                               user_id: @user.id
+                                             }
+                                  }
+    end
+    assert_redirected_to posts_path
+  end
+
+   # testing for redirects when trying to create a comments if not logged in
+  test 'redirects when trying to create a post if not logged in' do
+    assert_no_difference 'Comment.count' do
```

Comment:
Here, what value do we get the comment.count?

Reply:
sir this means if user are not log in so count of comments is same because no one comments without log in